controllers/compassmanager_controller.go

```diff
@@ -125,232 +126,124 @@ func (cm *CompassManagerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
+		_ = cluster.SetCompassMappingStatus(req.NamespacedName, true, false)
 		cm.Log.Warnf("Failed to configure Compass Runtime Agent for Kyma resource %s: %v.", req.Name, err)
 		return ctrl.Result{RequeueAfter: cm.requeueTime}, err
 	}
-	cm.Log.Infof("Compass Runtime Agent for Runtime %s configured.", compassRuntimeID)
 
-	return ctrl.Result{}, nil
-}
+	_ = cluster.SetCompassMappingStatus(req.NamespacedName, true, true)
 
-func (cm *CompassManagerReconciler) getKubeconfig(kymaName string) (string, error) {
-	secretList := &corev1.SecretList{}
-	labelSelector := labels.SelectorFromSet(map[string]string{
-		LabelKymaName: kymaName,
-	})
-
-	err := cm.Client.List(context.Background(), secretList, &client.ListOptions{
-		LabelSelector: labelSelector,
-	})
-
-	if err != nil {
-		return "", err
-	}
-
-	if len(secretList.Items) == 0 {
-		return "", nil
-	}
-	secret := &secretList.Items[0]
-
-	return string(secret.Data[KubeconfigKey]), nil
-}
-
-func (cm *CompassManagerReconciler) getKymaCR(objKey types.NamespacedName) (kyma.Kyma, error) {
-	instance := kyma.Kyma{}
-
-	err := cm.Client.Get(context.Background(), objKey, &instance)
-	if err != nil {
-		return instance, err
-	}
-
-	if instance.Labels == nil {
-		instance.Labels = make(map[string]string)
-	}
-	if instance.Annotations == nil {
-		instance.Annotations = make(map[string]string)
-	}
-
-	return instance, nil
-}
-
-func (cm *CompassManagerReconciler) upsertCompassMappingResource(compassRuntimeID, namespace string, kymaLabels map[string]string) error {
-	kymaName := kymaLabels[LabelKymaName]
-	compassMapping := &v1beta1.CompassManagerMapping{}
-	compassMapping.Name = kymaName
-	compassMapping.Namespace = namespace
-
-	compassMappingLabels := make(map[string]string)
-	compassMappingLabels[LabelKymaName] = kymaLabels[LabelKymaName]
-	compassMappingLabels[LabelComppassID] = compassRuntimeID
-	compassMappingLabels[LabelGlobalAccountID] = kymaLabels[LabelGlobalAccountID]
-	compassMappingLabels[LabelSubaccountID] = kymaLabels[LabelSubaccountID]
-	compassMappingLabels[LabelManagedBy] = "compass-manager"
-
-	compassMapping.SetLabels(compassMappingLabels)
-
-	key := types.NamespacedName{
-		Name:      kymaName,
-		Namespace: namespace,
-	}
-
-	existingMapping := v1beta1.CompassManagerMapping{}
-	// TODOs add retry for upsert logic
-	err := cm.Client.Get(context.TODO(), key, &existingMapping)
-	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			return cm.Client.Create(context.Background(), compassMapping)
-		}
-	}
+	cm.Log.Infof("Compass Runtime Agent for Runtime %s configured.", compassRuntimeID)
 
-	existingMapping.SetLabels(compassMappingLabels)
-	return cm.Client.Update(context.TODO(), &existingMapping)
+	return ctrl.Result{}, nil
 }
 
-func (cm *CompassManagerReconciler) getRuntimeIDFromCompassMapping(kymaName, namespace string) (string, error) {
-	mappingList := &v1beta1.CompassManagerMappingList{}
-	labelSelector := labels.SelectorFromSet(map[string]string{
-		LabelKymaName: kymaName,
-	})
```

Comment:
Is this comment still relevant? The definition of `setCompassMappingStatus` function is not here. See line 500